### PR TITLE
Models Validations, and Unit Specs Addition

### PR DESCRIPTION
**Pull request by mohamedSabry0**
**Key Enhancements:**

- Implemented validations for models as specified:
  - **User Model:**
    - Validated that the 'Name' attribute cannot be blank.
    - Ensured 'PostsCounter' is an integer greater than or equal to zero.
  - **Post Model:**
    - Enforced validation to disallow blank 'Title' entries.
    - Limited 'Title' to a maximum of 250 characters.
    - Validated 'CommentsCounter' and 'LikesCounter' as integers greater than or equal to zero.

- Added comprehensive unit specs covering all model methods and validations, ensuring code robustness and reliability.

Your review of these enhancements is appreciated for ensuring adherence to specifications and improving overall code quality.

### CODE REVIEW

File: spec/models/comment_spec.rb
Comment (Efiamotu-1) [inline spec/models/comment_spec.rb:1]: - [ ] All of your tests aren't running and i get the below error showing up for every spec you created, it shows that your code syntax is incorrect and i would advice you fix these errors, confirm that your tests works perfectly then resubmit. :+1: 
brun
below is an example code snippet that can assist you in creating tests for your comment_spec :sparkles: 
```ruby
require 'rails_helper'

RSpec.describe Comment, type: :model do
  context 'Testing the comment model' do
    it 'is valid with valid attributes' do
      @user01 = User.create(name: 'myogspike', photo: 'https://image.com/01.jpg', bio: 'Born in USA', posts_counter: 0)
      @post01 = Post.create(author: @user01, title: 'dddHello Mark', text: 'What about you ?', likes_counter: 0,
                            comments_counter: 0)
      expect(@post01).to be_valid
    end

    it 'update_comments_counter' do
      @user01 = User.create(name: 'pires', photo: 'https://image.com/01.jpg', bio: 'Born in USA', posts_counter: 0)
      @post01 = Post.create(author: @user01, title: 'Hello Mark', text: 'What about you ?', likes_counter: 0,
                            comments_counter: 0)
      Comment.create(user: @user01, post: @post01, text: 'Now working in dev')
      expect(@post01.comments_counter).to eq 1
    end
  end
end
```

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @mohamedSabry0, 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+  end
+
+  describe '#update_comments_count_on_create' do
+    it 'updates the comments count on create' do
+      user1 = User.create(name: 'John Doe')
+      post1 = Post.create(author: user, title: 'New post')
+      expect do
+        Comment.create(user: user1, post: post1, text: 'New comment')
+      end.to change { post.comments_counter }.by(1)
+    end
```
Comment (shahadat3669) [inline spec/models/comment_spec.rb]: 👏 Fantastic effort on your comprehensive testing! 👏 However, it seems there's a specific test case that is not passing. Let's focus on improving it:

- [ ] **Reviewing and Updating Test Case:** The test case in question doesn't pass, and it appears there might be an issue with the reference to the variable `user` in line number 12. Please review your code, and if the variable should indeed be `user`, ensure that it is defined correctly.
image

By addressing this issue, you'll enhance the accuracy of your test cases. Keep up the meticulous testing! 🚀🧪

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+  end
+
+  describe '#update_likes_count_on_create' do
+    it 'updates the likes count on create' do
+      user1 = User.create(name: 'John Doe')
+      post1 = Post.create(author: user, title: 'New post')
+      expect do
+        Like.create(user: user1, post: post1)
+      end.to change { post.likes_counter }.by(1)
+    end
```
Comment (shahadat3669) [inline spec/models/like_spec.rb]: 👏 Fantastic effort on your comprehensive testing! 👏 However, it seems there's a specific test case that is not passing. Let's focus on improving it:

- [ ] **Reviewing and Updating Test Case:** The test case in question doesn't pass, and it appears there might be an issue with the reference to the variable `user` in line number 12. Please review your code, and if the variable should indeed be `user`, ensure that it is defined correctly.
image

By addressing this issue, you'll enhance the accuracy of your test cases. Keep up the meticulous testing! 🚀🧪

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  describe '#five_most_recent_comments' do
+    it 'returns the five most recent comments for the post' do
+      user1 = User.create(name: 'John Doe')
+      post1 = Post.create(author: user, title: 'New post')
+      comments = []
+      10.times do |i|
+        comments.push(Comment.create!(user: user1, post: post1, text: "Comment #{i}"))
+      end
+      expect(post.five_most_recent_comments).to eq(comments.last(5).reverse)
+    end
```
Comment (shahadat3669) [inline spec/models/post_spec.rb]: 👍 Good effort on your work! However, it seems there's room for improvement, specifically regarding a test case that doesn't pass:

- [ ] **Reviewing and Updating Test Case:** It appears that the test case is encountering an issue, possibly related to the reference to the variable `user` in line number 20. Please carefully check your code, and if the variable should indeed be `user`, make sure it is defined correctly.
image

By addressing this issue, you'll enhance the reliability of your test cases. Keep up the good work and strive for test case accuracy! 🚀🧪

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @mohamedSabry0, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ User model has proper validations.
✅ Has unit tests for `User`, `Post`, `Comment`, and `Like` model 💯


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by shahadat3669 — APPROVED** [review]
Hi @mohamedSabry0, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ User model has proper validations
✅ `Post` model has proper validations
✅ Has unit tests for `User`, `Post`, `Comment`, and `Like` model

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**